### PR TITLE
Server game integration

### PR DESCRIPTION
**Pull request by Codrux2200**
## Description
implementation du score

## Lien tache Jira
https://leomaman.atlassian.net/browse/TYPE-126

## Changement
changement des 2 coté a de multiples endroit

## Summary by CodeRabbit

- **New Features**
  - Introduced a scoring system in the game, with scores increasing by 10 for each entity removed due to a player bullet.
  - Added a new entity to display the score in the game scene.

- **Bug Fixes**
  - Improved shooting logic in the game.

- **Refactor**
  - Modified the `removeEntitiesToDestroy` function to return an updated score.
  - Added a new handler for the `SCORE` packet type.

- **Style**
  - Cleaned up code by removing empty lines.

### CODE REVIEW

File: server/ecs/models/EnemyEntity.cpp [hunk 1]
```diff
@@ -12,14 +12,14 @@
 #include "PositionComponent.hpp"
 #include "PacketManager.hpp"
 #include "EnemyBulletComponent.hpp"
+#include <iostream>
```
Comment (christophechr) [inline server/ecs/models/EnemyEntity.cpp:15]: Not used

File: server/ecs/models/PlayerBullet.cpp [hunk 2]
```diff
@@ -13,7 +13,6 @@
 #include "PlayerBulletComponent.hpp"
 #include "PositionComponent.hpp"
 #include "VelocityComponent.hpp"
-
```
Comment (christophechr) [inline server/ecs/models/PlayerBullet.cpp:16]: Don't need to remove space, making easier to read the header file

**Review by christophechr — CHANGES_REQUESTED** [review]

File: client/ecs/Core.cpp [hunk 3]
```diff
@@ -56,6 +56,16 @@ void ECS::Core::_initHandlers(Network::PacketManager &packetManager)
...
     packetManager.REGISTER_HANDLER(Network::PacketType::BOSS_STATE, &ECS::Core::_handlerBossState);
+    packetManager.REGISTER_HANDLER(Network::PacketType::SCORE, &ECS::Core::_handlerScore);
+}
+
+void ECS::Core::_handlerScore(Network::Packet &packet, const udp::endpoint &endpoint){
+    _score = packet.scoreData.Score;
+    for (auto entity : sceneManager.getCurrentScene()->entitiesList){
+        if (entity->getComponent<ECS::TextComponent>() != nullptr && entity->getComponent<ECS::TextComponent>()->getString().find("Score") != std::string::npos) {
+            entity->getComponent<ECS::TextComponent>()->setText("Score " + std::to_string(_score));
+        }
```
Comment (raphaelMrci) [inline client/ecs/Core.cpp:67]: Replace by a find ScoreComponent

File: client/ecs/components/ScoreBoardComponent.cpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+#include "ScoreBoardComponent.hpp"
+
+namespace ECS{
+
+    ScoreBoardComponent::ScoreBoardComponent()
+    {
+    }
+
+    ScoreBoardComponent::~ScoreBoardComponent()
```
Comment (christophechr) [inline client/ecs/components/ScoreBoardComponent.cpp:16]: In the .hpp, just set the dtor as default

File: client/ecs/components/ScoreBoardComponent.hpp [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+#define SCOREBOARDCOMPONENT_HPP_
+#include <iostream>
+#include <memory>
+#include "AComponent.hpp"
+
+namespace ECS{
+    class ScoreBoardComponent : public AComponent{
+        public:
+            ScoreBoardComponent();
+            ~ScoreBoardComponent();
```
Comment (christophechr) [inline client/ecs/components/ScoreBoardComponent.hpp:18]: ~ScoreBoardComponent() = default;

**Review by christophechr — CHANGES_REQUESTED** [review]

**Review by raphaelMrci — APPROVED** [review]

**Review by christophechr — APPROVED** [review]

**Pull request merged**